### PR TITLE
Add TRC unit tests to concord-bft

### PR DESCRIPTION
**Pull request by shgandhi**
This PR adds a commit that adds TRC unit tests. This PR also includes commits with minor fixes.
Note that`trace_context_test.cpp` has not yet been added. Once it is verified that trace_contexts are safely propagated to/from TRC and concord with the move to open-source, the test will be added.

### CODE REVIEW

**Comment by upshaw-alex** [comment]
Thank you very much for adding the tests to the open source. These changes look good to me for the most part, though I am concerned about the checks that have failed.

File: client/thin-replica-client/test/CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+		      thin_replica_client
+					GTest::Main
+  				GTest::GTest)
+
+add_test(NAME trc_hash_tests COMMAND trc_hash_tests)
+add_executable(trc_hash_tests trc_hash_test.cpp)
+target_include_directories(trc_hash_tests PRIVATE ../src)
+target_link_libraries(trc_hash_tests
+                      thin_replica_client
+											GTest::Main
```
Comment (robem) [inline client/thin-replica-client/test/CMakeLists.txt:54]: Nit: Tabs and spaces in this file.

**Review by robem — APPROVED** [review]
Great logical commit, easy to follow. Thanks! 💪

File: client/thin-replica-client/test/CMakeLists.txt (hunk 1 above)
Comment (shgandhi) [reply client/thin-replica-client/test/CMakeLists.txt:53]: Thanks for catching that @robem, updated.

**Review by upshaw-alex — APPROVED** [review]
Thank you for the clean up you are doing to bring these tests up to Concord-BFT's coding standards.

**Pull request merged**